src/sidebar/components/hypothesis-app.js

```diff
@@ -142,6 +153,13 @@ function HypothesisAppController(
       $rootScope.$emit(events.ANNOTATION_DELETED, draft);
     });
     drafts.discard();
+
+    if (serviceConfig(settings)) {
+      // Let the host page handle the signup request
+      bridge.call(bridgeEvents.LOGOUT_REQUESTED);
+      return;
+    }
```

Comment:
The code for the help button (above) has to handle sending the bridge event if a help callback is registered, or showing the help panel if one isn't. This code for the log out button doesn't check whether a log out callback is registered, it just blindly sends the bridge event, there is no procedure for logging out of a third-party account with no log out callback and also the log in control component ensures that no log out button would be displayed in that case anyway.